src/formElementPlugins/fileUpload/src/ApplicationView.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { useEffect, useState, useRef } from 'react'
+import { Button, Icon, Grid, List, Image, Message, Segment, Loader } from 'semantic-ui-react'
+import { ApplicationViewProps } from '../../types'
+import strings from '../constants'
+import config from '../../../config.json'
+import { useUserState } from '../../../contexts/UserState'
+import { useRouter } from '../../../utils/hooks/useRouter'
+
+interface FileInfo {
+  filename: string
+  fileUrl: string
+  thumbnailUrl: string
+  mimetype: string
+}
+
+interface FileLoading {
+  filename: string
+  loading: boolean
+}
+
+interface FileError {
+  filename: string
+  error: boolean
+  errorMessage: string
+}
+
+const host = config.serverREST
+const { uploadEndpoint } = config
+
+const ApplicationView: React.FC<ApplicationViewProps> = ({
+  code,
+  parameters,
+  onUpdate,
+  value,
+  setValue,
+  isEditable,
+  // currentResponse,
+  validationState,
+  onSave,
+  Markdown,
+  initialValue,
+}) => {
+  const { label, description, fileCountLimit, fileExtensions, fileSizeLimit } = parameters
+
+  // These values required for file upload query parameters:
+  const {
```

Comment:
We should probably clarify exactly what the front-end plugins have access to, and how they get it. It's pretty clear in the back-end Actions now, but front-end is a bit vague. For example, I've used the global "config.json" file to get the server host details, and there isn't really any way to evaluate this as a parameter expression. It would need to passed in separately from the "parameters", which are evaluator expressions.

Perhaps we need a "globalData" object that gets passed down into all plugins, similar to the "applicationData" object in the back-end? That seems like the most straightforward solution.

It would contain (just off the top of my head):
```
{
 user: (id, username, etc)
 application: (id, serial, name, etc.)
 config: (whole config.json object)
 ... (more as we think of them/need them)
}
```
Point is, these values would need to be "static" and don't really change depending on the application state or plugin type, as opposed to "parameters" which are inherently dynamic.

Reply:
Made an issue to discuss this: #458 
For now we'll leave as is?

Reply:
Thanks for the summary, will leave it for next week